dandi/support/digests.py

```diff
@@ -103,22 +107,93 @@ def get_zarr_checksum(
 ) -> str:
```

Comment:
please add a docstring describing the purpose of `basepath` and `known` since hard to impossible to guess